dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassDumperTest.java

```diff
@@ -313,4 +314,26 @@ public void testIsUnusedClassSymbolReference_classSymbolReferenceNotFound()
                   + " constant pool.");
     }
   }
+
+  @Test
+  public void testLoadJavaClass_classNotInRoot() throws IOException, URISyntaxException {
+    // This JAR file contains com.google.firestore.v1beta1.FirestoreGrpc under BOOT-INF/classes.
+    // Spring-cloud-gcp's spring-cloud-gcp-kotlin-app-sample module generates such file.
+    Path path = absolutePathOfResource("testdata/dummy-boot-inf-prefix.jar");
+    ClassDumper classDumper = ClassDumper.create(ImmutableList.of(path));
+
+    SymbolReferenceMaps symbolReferences = classDumper.findSymbolReferences();
+    symbolReferences
+        .getClassToClassSymbols()
+        .forEach(
+            (classFile, classSymbol) -> {
+              try {
+                classDumper.loadJavaClass(classFile.getClassName());
+              } catch (ClassNotFoundException ex) {
+                fail(
+                    "The class in findSymbolReferences should be able to load again: "
```

Comment:
I'm not sure what this message means

Reply:
Simplified the test and removed the message.